client.lua

```diff
@@ -731,6 +731,35 @@ function SpawnPeds()
 				TaskStartScenarioInPlace(spawnedped, v.scenario, 0, true)
 			end
 
+			if v.pedrelations and type(v.pedrelations.groupname) == 'string' then
+				local pedgrouphash = joaat(v.pedrelations.groupname)
+
+				if not DoesRelationshipGroupExist(pedgrouphash) then
+					AddRelationshipGroup(v.pedrelations.groupname)
+				end
+
+				SetPedRelationshipGroupHash(spawnedped, pedgrouphash)
+				if v.pedrelations.toplayer then
+					SetRelationshipBetweenGroups(v.pedrelations.toplayer, pedgrouphash, joaat('PLAYER'))
+				end
+
+				if v.pedrelations.toowngroup then
+					SetRelationshipBetweenGroups(v.pedrelations.toowngroup, pedgrouphash, pedgrouphash)
+				end
+			else
+				error(v.pedrelations.groupname .. ' is not a string')
```

Comment:
This will throw an error inside the already made error because it can't recognize v.pedrelations if it doesn't exist, put this failsafe in the if statement like this instead
`if type(v.pedrelations.groupname) ~= 'string' then return error(v.pedrelations.groupname .. ' is not a string') end` and then you can remove the else statement and the type check inside the original if statement to prevent errors